enterprise/static/enterprise/grant_data_sharing_permissions.js

```diff
@@ -1,16 +1,85 @@
+  }
+}
+
+function hideBackgroundFromTabIndex() {
+    $(".background-input").attr("tabindex", "-1");
+    $(".consent-container").attr("aria-hidden", "true");
+}
+
+function showBackgroundInTabIndex() {
+    $(".background-input").attr("tabindex", "0");
+    $(".consent-container").attr("aria-hidden", "false");
+}
+
 (function() {
-    'use strict';
+    "use strict";
 
     $(document).ready(function() {
-        $('#data-sharing').submit(function(event) {
-            var $dataSharingConsentCheckbox = $('#register-data_sharing_consent');
-            var warningMessage = $dataSharingConsentCheckbox.data('warningMessage');
-
-            if (!$dataSharingConsentCheckbox.is(':checked') && warningMessage) {
-                // eslint-disable-next-line no-alert
-                if (!window.confirm(warningMessage)) {
-                    event.preventDefault();
-                }
+        $("#failure-link").click(function (event) {
+            $("#consent-confirmation-modal").show();
+            $("#modal-close-button").focus();
+            hideBackgroundFromTabIndex();
+        });
+        $("#consent-policy-dropdown-bar").click(function (event) {
+            togglePolicy();
+        });
+        $(".policy-dropdown-link").click(function (event) {
+            togglePolicy(true);
+        });
+        $("#modal-no-consent-button").click(function (event) {
+            $("#data-consent-checkbox").attr("checked", false);
+            $("#data-sharing").submit();
+        });
+        $("#review-policy-link").click(function (event) {
+            $("#consent-confirmation-modal").hide();
+            showBackgroundInTabIndex();
+            togglePolicy(true);
+        });
+        $("#consent-confirmation-modal").click(function (event) {
+            $("#consent-confirmation-modal").hide();
+            showBackgroundInTabIndex();
+        });
+        $("#data-consent-checkbox").change(function (event) {
+            $("#consent-button").attr("disabled", !(this.checked));
+        });
+        $(document).keypress(function (event) {
+            if ((event.keyCode === 27) && ($("#consent-confirmation-modal").is(":visible"))) {
+                $("#consent-confirmation-modal").hide();
+                showBackgroundInTabIndex();
+                event.preventDefault();
+            }
+        });
+        $("#review-policy-link").keydown(function (event) {
+            if (event.keyCode == 9 && !event.shiftKey){
+                event.preventDefault();
+                $("#modal-close-button").focus();
+            }
+        });
+        $("#modal-close-button").keydown(function (event) {
+            if (event.keyCode == 9 && event.shiftKey){
+                event.preventDefault();
+                $("#review-policy-link").focus();
+            }
+        });
+        analytics.track("data_consent_form_shown");
+        analytics.trackForm($("#data-sharing"), function () {
```

Comment:
Ok. So all consent requests will come from /enterprise/grant_data_sharing_permissions, and the characteristics of the consent request will be defined via query params.

Are there always a fixed number of query params (course_id, failure_url, next) or is this variable? It'd be great to have those query params separated into their own fields.

Use case will be similar to the below image (different event, similar concept). I will want to pull via Google Analytics the overall event count (consent shown, accepted, denied) along with various properties of that consent. Having it be such a long `search` field will make it difficult to aggregate since the data will end up being sparse.

![image](https://cloud.githubusercontent.com/assets/6895019/24291502/9849cc76-105f-11e7-879c-8ed301a6617c.png)